src/distributed-callback-queue.js

```diff
@@ -209,14 +222,21 @@ class DistributedCallbackQueue {
 
     let onCompleted;
     try {
-      onCompleted = await this.push(suffix, onJobCompleted, timeout);
+      const pushPromise = this.push(suffix, onJobCompleted, timeout);
+      onCompleted = await Promise.race([
+        pushPromise,
+        jobAbortPromise,
+      ]);
+
+      jobAbortPromise = undefined;
     } catch (err) {
+      jobAbortPromise = undefined; // finally {} is too late
```

Comment:
lets add `jobAbortReject = undefined` to be extra sure to both catch/resolve chains and I believe this is good to be merged!

Reply:
change made & pushed